assessments/projects/custodian/joint-review.md

```diff
@@ -0,0 +1,923 @@
+| passwd               |                  |          |                          |               |                                |
+|                      |                  |          |                          |               |                                |
+|                      |                  |          |                          |               |                                |
++----------------------+------------------+          +--------------------------+---------------+--------------------------------+
+| python-pip-whl       | CVE-2020-26137   |          | 20.0.2-5ubuntu1.6        |               | python-urllib3: CRLF injection |
+|                      |                  |          |                          |               | via HTTP request method        |
++----------------------+------------------+          +--------------------------+---------------+--------------------------------+
+| python3.8            | CVE-2021-23336   |          | 3.8.10-0ubuntu1~20.04    |               | python: Web cache poisoning    |
+|                      |                  |          |                          |               | via urllib.parse.parse_qsl and |
+|                      |                  |          |                          |               | urllib.parse.parse_qs by using |
+|                      |                  |          |                          |               | a semicolon...                 |
++----------------------+                  +          +                          +---------------+                                +
+| python3.8-minimal    |                  |          |                          |               |                                |
+|                      |                  |          |                          |               |                                |
+|                      |                  |          |                          |               |                                |
+|                      |                  |          |                          |               |                                |
++----------------------+                  +          +                          +---------------+                                +
+| python3.8-venv       |                  |          |                          |               |                                |
+|                      |                  |          |                          |               |                                |
+|                      |                  |          |                          |               |                                |
+|                      |                  |          |                          |               |                                |
++----------------------+------------------+----------+--------------------------+---------------+--------------------------------+
+```
```

Comment:
Do we need the `HIGH` vulnerabilities to be resolved or at least require the c7n team to open issues for them and link them here ?

Reply:
These are operating system library level considerations not per se the project's responsibility to fix, but the distribution vendor, ie a libc high issue is something i would expect the os vendor to triage and fix as needed. This is also specific to a distribution channel, and immaterial for those that install the project's software directly into their own OS / container base. For direct application dependency graphs concerns we already address those pro actively (dependabot alerts etc).

Reply:
these were from the Github action docker scan - if the project is releasing/maintaining a docker image I think the user expectation is they should be fixing high security risks in the image as good hygiene.

maybe best to just do source releases? or clearly document that the image is not secure?

Reply:
This feels like a misunderstanding of operating system docker images, and what severity levels correspond to a need to immediate triage. If we take any production docker image out there using an operating system, they all have these sorts of cve nits wrt to `high` severity scoring. Take anything from redis to postgresql, (trivy image redis:latest etc), that doesn't make them insecure or high risk (trivy output for both here https://gist.github.com/cbdd697ce7f8ad0a40896c0a081c1a7c). The most relevant cves are those deemed critical, but all of operating system cves are triaged and dealt with by the security team of the respective distros which in many cases are professional security teams focused on exactly this topic. Moreover unlike postgresql or redis, we're talking about software in custodian that doesn't have an open port, its literally a cli, not a daemon or service running in a container.

Reply:
My original comment was related to direct dependancies of c7n. @kapilt I think you mentioned you have a process to handle those iiuc.

Reply:
for custodian's dependency graph, we rely on dependabot alerts https://docs.github.com/en/code-security/supply-chain-security/managing-vulnerabilities-in-your-projects-dependencies/about-alerts-for-vulnerable-dependencies against our poetry.lock files which contain the version/hash dependency pins for the project.  that alert is a prompt for us to update our dependency graph. our dep graph update process is automated and documented https://cloudcustodian.io/docs/developer/packaging.html and done regularly independent of vulnerabilities as a prelude to our release process.

Reply:
also to be specific there are zero cves extant on our application dependency graph at the moment.